src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +28,31 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public int hashCode() {
+        int result = ((color == null) ? 0 : color.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) obj;
+        if ((width == null && length == null && color == null)) {
+            return true;
+        }
+        if ((width == rectangle.width) && (length == rectangle.length)
+                && (color.equals(rectangle.getColor()))) {
+            return true;
+        }
+        return true;
```

Comment:
should we return true if no conditions from above gave us true?

Reply:
No